### PR TITLE
Feature/yeelight

### PR DESCRIPTION
**Pull request by anakod**
Initial support for great yeelight smart WiFi bulbs.
http://www.yeelight.com/en_US/product/wifi-led-c

More information about API below (with examples):
http://www.yeelight.com/en_US/developer
http://www.yeelight.com/download/Yeelight_Inter-Operation_Spec.pdf

This branch was successfully tested on hardware, of course.

### CODE REVIEW

File: Sming/Services/Yeelight/YeelightBulb.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include "IPAddress.h"
+class TcpClient;
+
+#ifndef SERVICES_YEELIGHTBULB_H_
+#define SERVICES_YEELIGHTBULB_H_
+
+class YeelightBulb
+{
+public:
+	YeelightBulb(IPAddress addr);
```
Comment (slaff) [inline Sming/Services/Yeelight/YeelightBulb.h]: @anakod Can you please add documentation comments? Take a look at [that document](https://github.com/SmingHub/Sming/wiki/Documenting-the-API) if you need guiding lines.

File: samples/TcpClient_NarodMon/app/application.cpp [hunk 2]
```diff
@@ -51,6 +51,7 @@ bool nmOnReceive(TcpClient& client, char *buf, int size)
 	// debug msg
 	debugf("nmOnReceive");
 	debugf("%s", buf);
+	return true;
```
Comment (slaff) [inline samples/TcpClient_NarodMon/app/application.cpp]: That change should be part of another PR related to fixes in the TcpClient_NarodMon sample.

Comment (anakod) [inline samples/TcpClient_NarodMon/app/application.cpp]: Ok, I will revert it. Agreed it isn't part of this "logical" PR, but it's here because this TCP example have led me to the error in my new library callback in the begining.

Comment (anakod) [inline samples/TcpClient_NarodMon/app/application.cpp]: For reference: https://github.com/SmingHub/Sming/blob/develop/Sming/SmingCore/Network/TcpClient.cpp#L126

File: Sming/Services/Yeelight/YeelightBulb.h (hunk 1 above)
Comment (anakod) [inline Sming/Services/Yeelight/YeelightBulb.h]: Ok, but I'm not sure what should I write to docs:
on - is on,
off - is off,
setRGB - set RGB color
and etc..

I've added description for all non-trivial points what can be useful for developers.

**Comment by slaff** [comment]
@anakod Thanks! If you are ready with your changes I will take care to merge the PR once travis reports success in SDK 1.4 and SDK 1.5 builds on Linux (MacOS builds for the moment take forever to start).

**Comment by anakod** [comment]
Yes, I'm ready, lets merge and go forward.
